sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -494,44 +494,54 @@ object DateTimeUtils {
   }
 
   /**
-   * Returns the month value for the given date. The date is expressed in days
-   * since 1.1.1970. January is month 1.
+   * Split date (expressed in days since 1.1.1970) into four fields:
+   * year, month (Jan is Month 1), dayInMonth, daysToMonthEnd (0 if it's last day of month).
    */
-  def getMonth(date: Int): Int = {
+  def splitDate(date: Int): (Int, Int, Int, Int) = {
     var (year, dayInYear) = getYearAndDayInYear(date)
-    if (isLeapYear(year)) {
-      if (dayInYear == 60) {
-        return 2
-      } else if (dayInYear > 60) {
-        dayInYear = dayInYear - 1
+    val isLeap = isLeapYear(year)
+    val (month, dayOfMonth, daysToMonthEnd) =
```

Comment:
Ah, I've already removed this temporary tuple when I was benchmarking. just like @davies 's code above.